wg-api/spec-documents/scroll-view.md

```diff
@@ -0,0 +1,178 @@
+# Scroll View
+
+## Summary
+
+This spec proposes to add an experimental native ScrollView component to Electron.
+
+## Platforms
+* macOS 
+* Windows
+* Linux
+
+## Impetus
+
+It is currently not possible to add scrolling functionality around BrowserViews:
+
+* Capture mouse / touch events
+* Reposition BrowserViews accordingly
+
+More details of the use-case and challenges related to workarounds: [#32751](https://github.com/electron/electron/issues/32751)
+
+![](https://user-images.githubusercontent.com/2891206/150981006-84fe796e-60bd-43c0-952d-831d1bfd9998.gif)
+
+## Usage Example
+
+```js
+const { BaseWindow, ContainerView, ScrollView, BrowserView } = require("electron");
+
+const window = new BaseWindow({ width: 1000, height: 500 })
+
+const scroll = new ScrollView()
+scroll.setBounds({ width: 1000, height: 500 })
+
+const scrollContent = new ContainerView()
+scrollContent.setBounds({ width: 2000, height: 500 })
+
+const browserView1 = new BrowserView()
+browserView1.setBounds({ width: 1000, height: 500 })
+scrollContent.addBrowserView(browserView1)
+
+const browserView2 = new BrowserView()
+browserView2.setBounds({ width: 1000, height: 500, x: 1000 })
+scrollContent.addBrowserView(browserView2)
+
+scroll.setContentView(scrollContent);
+win.setContainerView(scroll);
+```
+
+
+## API Design
+
+In order to achieve the example above, we've made following changes to the API
+
+### BaseView
+
+A `BaseView` is a rectangle within the views View hierarchy. It is the base
+class for all all different views. 
+
+Process: **Main**
+
+`BaseView` is an EventEmitter
+
+
+#### Static Methods
+
+* `BaseView.getAllViews()`
```

Comment:
What's the intended use case for this method?

Reply:
It returns all ScrollView's created - added for consistency (BaseView.getAllViews() -> ScrollView.getAllViews())

Reply:
I think this is kind of a misfeature in BaseView, I don't see the value in maintaining this list.

Reply:
We can remove this function from the proposal, no prob. The main goal for the API is to have the possibility to build hierarchy of views - ScrollView is the most important part.

Reply:
Cool, sounds good. Let's not add functions that have no clear purpose :)

Reply:
Removed - also did the small adjustment in the example :)